src/Propulsion.MessageDb/MessageDbSource.fs

```diff
@@ -13,6 +13,18 @@ module internal Npgsql =
         do! conn.OpenAsync(ct)
         return conn }
 
+
+module private Queries =
+    [<Literal>]
+    let getCategoryMessages =
+        "select position, type, data, metadata, id::uuid,
+           (metadata::jsonb->>'$correlationId')::text,
+           (metadata::jsonb->>'$causationId')::text,
+           time, stream_name, global_position
+         from get_category_messages($1, $2, $3);"
+    [<Literal>]
+    let getLastPosition = "select max(global_position) from messages where category(stream_name) = $1;"
```

Comment:
Found that this greatly reduced memory pressure on the messagedb store when at the tail